tests/test_key_validator.py

```
@@ -0,0 +1,137 @@
+            success, msg = check_connection("test")
+            assert not success
+            assert "RuntimeError" in msg
+            assert "boom" not in msg  # Details hidden from user
+
+
+class TestConnectionOpenAI:
+    def test_openai_success(self):
+        with patch.dict("qwed_new.providers.key_validator._TEST_HANDLERS", {
+            "openai": lambda key, url, model, t: (True, "Connected to OpenAI API.")
+        }):
+            success, msg = check_connection("openai", api_key="TEST_fake_key_not_real")
```

Ruff (0.15.6)

[warning] Unpacked variable `msg` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)